tests/flow/test_hnsw_parallel.py

```diff
@@ -208,10 +215,12 @@ def run_queries():
     t_query.start()
 
     [t.join() for t in [t_insert, t_query]]
+    assert parallel_index.index_size() == num_elements
+    assert parallel_index.check_integrity()
 
     # Measure recall - expect to get increased recall over time, since vectors are being inserted while queries
     # are running, and the ground truth is measured compared to the index that contains all the elements.
-    chunk_size = int(num_queries/5)
+    chunk_size = int(num_queries/2)
```

Comment:
how is this fixing flakiness?

Reply:
The queries are now divided into 2 batches. The first batch runs after 50% of the vectors are indexed, and the second batch will start running only after 75% of the vectors are indexed. Then, we compare the recall between the two batches (expecting the second one to have a higher recall). Before, there was no waiting for 75%, and we just tested the recall for 5 consecutive batches (which was prone to flakiness)

Reply:
I don't see how the chunks depend on each other
I understand that increasing the size of the index statically increase the probability to find the gt vectors, but it is not determinstic since the queries are different.
we might get a very high recall because the gt vectors of the first queries chunk were luckly already indexed in the 50% vectors,
but all the gt vectors answering the second queries chunk are added last, resulting in a low low recall for the second chunk.

I suggest running the same queries on all index sizes, and compare the `total_correct_cur_chunk` to the accumelated total correct (well, for 2 chunks this is the same as comparing it to `total_correct_prev_chunk`)

Does it make sense?

Reply:
taking a ride on this comment for a nitpick reminder :D 
remove `expected_parallel_rate ` from`test_parallel_insert_multi`